src/Mod/Part/App/TopoShape.cpp

```diff
@@ -1199,6 +1200,35 @@ bool TopoShape::getCenterOfGravity(Base::Vector3d& center) const
     return false;
 }
 
+bool TopoShape::getRotation(Base::Rotation& rotation) const
+{
+    if (_Shape.IsNull()) {
+        return false;
+    }
+
+    // Plane normal

```

Comment:
There is an infinite number of solutions to compute a rotation from two vectors. With this implementation the camera will correctly look at the found direction but the whole object in general is not aligned to the X or Y axis of the screen. It would be good to additionally determine an up vector and compute a unique rotation.

Reply:
I kinda moved this problem to View3DInventorViewer::alignToSelection(). I am still planning to improve the camera orientation and movement when aligning and to compute a unique orientation. But I don't think I can do that before the feature freeze (next week??). So if we want this feature in 1.0 then maybe the improvements should be in a separate PR.

Reply:
As you like.

What's basically needed is:
* if a planar face is selected check if it contains a straight edge as boundary and use its direction for the second vector
* if an edge is selected check if it's part of a planar face and use its normal for the second vector

Then use the class Base::CoordinateSystem to compute the rotation.